daemon/volumes_unix.go

```diff
@@ -316,7 +317,7 @@ func (daemon *Daemon) registerMountPoints(container *Container, hostConfig *runc
 			}
 
 			if len(cp.Source) == 0 {
-				v, err := daemon.createVolume(cp.Name, cp.Driver, nil)
+				v, err := daemon.createVolume(cp.Name, cp.Driver, nil, stringid.TruncateID(containerID))
```

Comment:
I rather not truncate IDs. There is no need for it and it might introduce bugs if people don't know they need to call these functions with a truncated ID. Please, remove all `stringid.TruncateID`.